### PR TITLE
Adds tests and deferred tasks for consistency tests.

### LINKED ISSUE #344: Unique items clean-up
Should we have a deferred action to verify that each student has at most one group per assignment, one final submission, and one user per email address? If so, how?

### PR DESCRIPTION
**Pull request by moowiz**
Fixes #344

### CODE REVIEW

File: server/app/utils.py [hunk 1]
```diff
@@ -264,15 +283,58 @@ def assign_submission(backup_id, submit):
...
+            for subm in lst:
+                subm.key.delete()
+
+def unique_group(user):
+    G = ModelProxy.Group
+    key_func = lambda group: group.assignment
+    groups = G.query(G.member == user.key).fetch()
+    for lst in sort_by_assignment(key_func, groups):
+        if len(lst) > 1:
+            # TODO(martinis, denero) figure out what to do
```
Comment (moowiz) [inline server/app/utils.py]: Log me, take a look when you find it.

File: server/app/utils.py [hunk 2]
```diff
@@ -264,15 +283,58 @@ def assign_submission(backup_id, submit):
...
+    FS = ModelProxy.FinalSubmission
+
+    key_func = lambda subm: subm.assignment
+    submissions = FS.query(FS.submitter == user.key).fetch()
+    for lst in sort_by_assignment(key_func, submissions):
+        if len(lst) > 1:
+            lst.sort(key=lambda subm: subm.server_time)
+            lst = lst[1:]
+            for subm in lst:
+                subm.key.delete()
```
Comment (moowiz) [inline server/app/utils.py]: Log me, take a look when you find it.

**Comment by Some3141** [comment]
Is this ready for review?

**Comment by moowiz** [comment]
No.. We never really decided what to do with these things when we found them.

Or we did, and I didn't implement that. I forget...

**Comment by moowiz** [comment]
Actually... this already effectively got merged... If you look at `utils.py`, it already the checks in here, in the master branch.

**Comment by Some3141** [comment]
Can you port over the tests and we can close #344 with that.

**Comment by moowiz** [comment]
The tests were ported over as well. I guess I didn't clean this up when I should have. My bad...

**Pull request closed without merging**